### ISSUE TITLE
File Upload v2

### ISSUE THREAD

**Issue by MPatriot** [oqtane/oqtane.framework#3105]
This is more on previously discussed but perhaps a fresh start...

Ok, two websites(A & B).  Both Oqtane, uploaded / published via ftp from Visual Studio(I still don't see how else to do it), both on same server but in different application pools.

After some go around with the web host and when I first published anything about it here, we managed to get Website A uploading, at least to the root directory, after some read/write permission changes by the host and restarting the application pool.

Website B, though pretty much the same thing from my end and supposedly on their end does not allow me to upload even small files.  I have also created the host a user(host permissions just like mine) on both sites so they can test it themselves.  Here is what they tell me again, after multiple times of complete clean start and trying again:

We still get the below error, and your site is trying to upload data to the incorrect 'h:\' instead of your site root folder, it must be some code bug that not insert to the correct folder, you will need to debug it. And this error is not due to permission.

fail: Microsoft.AspNetCore.Server.Kestrel[13]
Connection id "0HMSK7BERA6OE", Request id "0HMSK7BERA6OE:00000003": An unhandled exception was thrown by the application.
System.IO.DirectoryNotFoundException: Could not find a part of the path 'h:\'.
at System.IO.FileSystem.CreateDirectory(String fullPath, Byte[] securityDescriptor)
at System.IO.Directory.CreateDirectory(String path)
at Oqtane.Controllers.FileController.CreateDirectory(String folderpath) in C:\Projects\IOFlowsMich\oqtane.framework\Oqtane.Server\Controllers\FileController.cs:line 680
at Oqtane.Controllers.FileController.UploadFile(String folder, IFormFile formfile) in C:\Projects\IOFlowsMich\oqtane.framework\Oqtane.Server\Controllers\FileController.cs:line 359

.................There is a lot more but perhaps that is enough, I can always attach a file with the whole thing

I don't know what to do, if such an error is occurring Oqtane doesn't catch it, it only logs that file upload may have failed or still be in process ect.

I contend that this isn't what it appears to them to be but if they could solve this fir the first site then they ought to be able to do the same for the second.

Any help or thoughts or even ways to respond to the host as we are now at an impasse.  They won't look at it anymore, they believe I am trying to save to the root of the drive?  sheesh!

Mark

**Comment by sbwalker** [comment]
Yesterday I merged a PR #3100 which will catch exceptions when the framework attempts to create a directory on the file system - it will capture the folder path and error message. This will be included in the 4.0.2 release. However I doubt if this is going to reveal anything more than you already know - basically your site is not configured with sufficient permissions to create new directories dynamically.

**Comment by MPatriot** [comment]
Thanks Shaun @sbwalker  I wonder why the error they are stubbornly pointing to only refers to the drive.  They insist that the program is attempting to write to the root of the drive rather than the directory it would be written to.  
the path I can see is '/'  then '/myapppooldirectory1/' then BlazorDebugProxy, Content, Packages, wwwroot ect  AND I can dublicate that for more app pools so '/myapppooldirectory2/ditto' 
I can save even a mp4 file in the first setup but not the second.  Its the same thing but this tech support guy sees that error and insists I am trying to say to the wrong directory?  Ugg  

How do I tell him to get over himself and stop being a block head: get it to work!  they did it before with the first.

Thanks

**Comment by sbwalker** [comment]
@MPatriot if you browse to the System Info option in the Admin Dashboard it will display the server folder paths for the Web root (wwwroot) and Content root folders. Oqtane is multi-tenant so each "site" uses a unique folder naming convention. The root folder for an Oqtane "site" is "private" which means that it is located in the Content root folder location ie. {ContentRoot}/Content/Tenants/1/Sites/1

**Comment by sbwalker** [comment]
@MPatriot once you have a chance to upgrade to 4.0.2 please let me know what the detailed log message says in the Event Log related to creating folders

**Comment by MPatriot** [comment]
@sbwalker Thanks Shaun!

Amazing thing...

I have been pretty busy (and frustrated with all the above) so I stopped messing with it since about the time of my last post 

Anyway, I noticed your comment, sent to my email, and I ftp published 4.02 to the site this evening 

Everything worked fine. I was able to upload a small png file and then a 10 minute long mp4 file to the root directory. The larger file did so the timeout / possible error message but it continued to show the progress bar and completed the upload 

Hmm, be interesting to know if you changed anything other than error checking. Like I mentioned this was already worked on the same server in a different application pool

Either way I really appreciate you taking the time to get out a new version so quickly.  I also noticed I didn't have to delete any .cmd files. This would also be nice to know if the host support guys changed things or if this was also something different in v 4.02?  (I could look at all of the above and get the answers but ai also thought it good to report all the successes thanks to you guys)

Thanks again

Mark
